hibernate-core/src/main/java/org/hibernate/property/access/internal/AccessStrategyHelper.java

```diff
@@ -82,92 +80,13 @@ public static AccessType getAccessType(Class<?> containerJavaType, String proper
 				&& field.isAnnotationPresent( Access.class )
 				&& !field.isAnnotationPresent( Transient.class )
 				&& !Modifier.isStatic( field.getModifiers() ) ) {
-			return AccessType.FIELD;
+			return getAccessTypeOrNull( field );
 		}
 
-		for ( Method method : containerClass.getDeclaredMethods() ) {
-			// if the method has parameters, skip it
-			if ( method.getParameterCount() != 0 ) {
-				continue;
-			}
-
-			// if the method is a "bridge", skip it
-			if ( method.isBridge() ) {
-				continue;
-			}
-
-			if ( method.isAnnotationPresent( Transient.class ) ) {
-				continue;
-			}
-
-			if ( Modifier.isStatic( method.getModifiers() ) ) {
-				continue;
-			}
-
-			final String methodName = method.getName();
-
-			// try "get"
-			if ( methodName.startsWith( "get" ) ) {
-				final String stemName = methodName.substring( 3 );
-				final String decapitalizedStemName = Introspector.decapitalize( stemName );
-				if ( stemName.equals( propertyName ) || decapitalizedStemName.equals( propertyName ) ) {
-					if ( method.isAnnotationPresent( Access.class ) ) {
-						return AccessType.PROPERTY;
-					}
-					else {
-						checkIsMethodVariant( containerClass, propertyName, method, stemName );
-					}
-				}
-			}
-
-			// if not "get", then try "is"
-			if ( methodName.startsWith( "is" ) ) {
-				final String stemName = methodName.substring( 2 );
-				String decapitalizedStemName = Introspector.decapitalize( stemName );
-				if ( stemName.equals( propertyName ) || decapitalizedStemName.equals( propertyName ) ) {
-					if ( method.isAnnotationPresent( Access.class ) ) {
-						return AccessType.PROPERTY;
-					}
-				}
-			}
-		}
-
-		return null;
-	}
-
-	private static void checkIsMethodVariant(
-			Class<?> containerClass,
-			String propertyName,
-			Method method,
-			String stemName) {
-		final Method isMethodVariant = findIsMethodVariant( containerClass, stemName );
-		if ( isMethodVariant == null ) {
-			return;
-		}
-
-		if ( !isMethodVariant.isAnnotationPresent( Access.class ) ) {
-			throw new MappingException(
-					String.format(
-							Locale.ROOT,
-							"Class '%s' declares both 'get' [%s] and 'is' [%s] variants of getter for property '%s'",
-							containerClass.getName(),
-							method.toString(),
-							isMethodVariant,
-							propertyName
-					)
-			);
-		}
-	}
-
-	public static @Nullable Method findIsMethodVariant(Class<?> containerClass, String stemName) {
-		// verify that the Class does not also define a method with the same stem name with 'is'
-		try {
-			final Method isMethod = containerClass.getDeclaredMethod( "is" + stemName );
-			if ( !Modifier.isStatic( isMethod.getModifiers() ) && isMethod.getAnnotation( Transient.class ) == null ) {
-				return isMethod;
-			}
-		}
-		catch (NoSuchMethodException ignore) {
+		Method getter = getterMethodOrNull( containerClass, propertyName );
+		if (getter != null
+			&& getter.isAnnotationPresent( Access.class )){
+			return AccessType.PROPERTY;
```

Comment:
We should also read the `Access` annotation value here IMO, instead of always choosing `PROPERTY`. Also, please respect Hibernate's [code style](https://hibernate.org/community/contribute/intellij-idea/#code-style).

```suggestion
		final Method getter = getterMethodOrNull( containerClass, propertyName );
		if ( getter != null && getter.isAnnotationPresent( Access.class ) ) {
			return castNonNull( method.getAnnotation( Access.class ) ).value();
```

Reply:
ok